File: app/src/main/java/com/example/themoviedb/person_details_screen/person_details_presenter/PersonDetailsPresenter.java
package com.example.themoviedb.person_details_screen.person_details_presenter;

import com.example.themoviedb.person_details_screen.person_details_model.AsyncFetchinImagesInterface;
import com.example.themoviedb.person_details_screen.person_details_model.PersonDetailsModel;
import com.example.themoviedb.person_details_screen.person_details_model.PersonDetailsModelInterface;
import com.example.themoviedb.person_details_screen.person_details_model.Profiles;
import com.example.themoviedb.person_details_screen.person_details_view.PersonDetailsViewInterface;

public class PersonDetailsPresenter {
    PersonDetailsViewInterface personDetailsViewInterface;
    PersonDetailsModelInterface personDetailsModelInterface ;

    public PersonDetailsPresenter(PersonDetailsViewInterface personDetailsViewInterface, PersonDetailsModelInterface personDetailsModelInterface) {
        this.personDetailsViewInterface = personDetailsViewInterface;
        this.personDetailsModelInterface = personDetailsModelInterface;
    }

    public void getPersonDetails() {
        personDetailsViewInterface.getPersonDetails();
    }

    public void fetchPersonImage(String s) {
        PersonDetailsModel.GetPhotos getPhotos=new PersonDetailsModel.GetPhotos(new AsyncFetchinImagesInterface() {

            @Override
            public void processFinish(Profiles personImage) {
                personDetailsViewInterface.setImage(personImage);
                personDetailsViewInterface.setImagesInAdapter();
            }
        });
        getPhotos.execute(s);
    }
}

File: app/src/main/java/com/example/themoviedb/person_details_screen/person_details_presenter/PersonDetailsPresenter.kt
package com.example.themoviedb.person_details_screen.person_details_presenter

import com.example.themoviedb.person_details_screen.person_details_model.AsyncFetchinImagesInterface
import com.example.themoviedb.person_details_screen.person_details_model.PersonDetailsModel
import com.example.themoviedb.person_details_screen.person_details_model.PersonDetailsModelInterface
import com.example.themoviedb.person_details_screen.person_details_model.Profiles
import com.example.themoviedb.person_details_screen.person_details_view.PersonDetailsViewInterface

class PersonDetailsPresenter(internal var personDetailsViewInterface: PersonDetailsViewInterface, internal var personDetailsModelInterface: PersonDetailsModelInterface) {

    fun getPersonDetails() {
        personDetailsViewInterface.getPersonDetails()
    }

    fun fetchPersonImage(s: String) {
        val getPhotos = PersonDetailsModel.GetPhotos(object : AsyncFetchinImagesInterface {
            override fun processFinish(personImage: Profiles) {
                personDetailsViewInterface.setImage(personImage)
                personDetailsViewInterface.setImagesInAdapter()
            }
        })
        getPhotos.execute(s)
    }
}
